client/dashboard/app/queries.ts

```diff
@@ -22,17 +22,17 @@ export function sitesQuery() {
 	};
 }
 
-export function siteQuery( siteId: string ) {
+export function siteQuery( siteIdOrSlug: string ) {
 	return {
-		queryKey: [ 'site', siteId ],
+		queryKey: [ 'site', siteIdOrSlug ],
 		queryFn: async () => {
 			// Site usually takes the longest, so kick it off first.
-			const sitePromise = fetchSite( siteId );
+			const sitePromise = fetchSite( siteIdOrSlug );
```

Comment:
Should we update the name of the argument in `data/index` to clarify that it's either `siteIdOrSlug` like here.

Reply:
Yep, this is done.

Reply:
I wonder if we should just require it be the slug if that's what we should be using everywhere as the unique ID.